10_rest/n0npax/cmd/puppy-server/main.go

```diff
@@ -0,0 +1,110 @@
+package main
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+
+	"gopkg.in/alecthomas/kingpin.v2"
+
+	puppy "github.com/anz-bank/go-course/10_rest/n0npax/pkg/puppy"
+	store "github.com/anz-bank/go-course/10_rest/n0npax/pkg/puppy/store"
+	"github.com/gin-gonic/gin"
+)
+
+var (
+	out io.Writer = os.Stderr
+)
+
+func main() {
+	config, err := parseArgs()
+	puppyFilePath, storeType, port := config.path, config.sType, config.port
+	exitOnError(err)
+
+	storer, err := createStorer(storeType)
+	exitOnError(err)
+
+	if puppyFilePath != "" {
+		file, err := os.Open(puppyFilePath)
+		exitOnError(err)
+		puppies, err := readPuppies(file)
+		exitOnError(err)
+		err = feedStore(storer, puppies)
+		exitOnError(err)
+	}
+	r := puppy.RestBackend(storer)
+	err = runSrv(r, port)
+	exitOnError(err)
+}
+
+type config struct {
+	path, sType string
+	port        int
+}
+
+func parseArgs() (config, error) {
+	var puppyFilePath, storeType string
+	var port int
+	kingpin.Flag("data", "path to file with puppies data").Short('d').ExistingFileVar(&puppyFilePath)
```

Comment:
Use `FileVar`, saves you the error checking below.

Reply:
good hint, thanks!

Reply:
fixed